packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -206,7 +207,14 @@ export function getLibraryName(
  * @returns
  */
 export function getWireName(context: TCGCContext, type: Type & { name: string }) {
-  // 1. Check if there's an encoded name
+  const xmlNameDecorator = ignoreDiagnostics(getTypeDecorators(context, type)).filter(
+    (x) => x.name === "TypeSpec.Xml.@name",
+  );
+  // 1. Check xml name encoding
+  if (xmlNameDecorator.length > 0) {
+    return xmlNameDecorator[0].arguments["name"];
```

Comment:
So we're thinking that there really shouldnt be a case where encodedName is used with this decorator, right? We'll always return what's in @xml.name